src/main/java/graphql/validation/ValidationError.java

```diff
@@ -10,18 +10,19 @@
 
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 @PublicApi
 public class ValidationError implements GraphQLError {
 
-    private final String message;
     private final List<SourceLocation> locations = new ArrayList<>();
     private final String description;
     private final ValidationErrorClassification validationErrorType;
-    private final List<String> queryPath;
-    private final Map<String, Object> extensions;
+    private final List<String> queryPath = new ArrayList<>();
+    private final Map<String, Object> extensions = new HashMap<>();
```

Comment:
One way to make this class consistent (in the field initializations part) is to initialize also the queryPath and the extensions (like I have done): this lead to the putAll problem that does not maintain the order of the keys. The other way is to remove this initialization. Let me know what do you think about it

Reply:
Make it a `ImmutableMap.of()` under the covers

`private final ImmutableMap<String, Object> extensions;`

and then do `this.extensions = ImmutableMap.of(builder.extensions)`

If it was already a immutable map then its a smart coy - if not it keeps order - ImmutableMaps are like `LinkedHashMap` in that sense

Reply:
This method `ImmutableMap.of(Map<> map)` does not exist so I used the ternary operator to initialize the `extensions` field with `.copyOf()` when the `builder.extension` is not null, otherwise the extensions is initialized with the `.of()` method